.github/workflows/ci.yml

```diff
@@ -265,94 +278,24 @@ jobs:
           command: build
           args: --verbose --release
 
-  clippy-deps:
-    name: Clippy (stable)
-    timeout-minutes: 30
+  cargo-deny:
+    name: Check deny.toml crate dependencies and validate licenses
     runs-on: ubuntu-latest
-    env:
-      CARGO_INCREMENTAL: 0
-      RUST_BACKTRACE: full
-
-    steps:
-      - uses: actions/checkout@v2.4.0
-        with:
-          persist-credentials: false
-
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          components: clippy
-          override: true
-
-      - uses: Swatinem/rust-cache@v1
-
-      - name: Show env vars
-        run: |
-            echo "Common env vars:"
-            echo "RUST_BACKTRACE=${{ env.RUST_BACKTRACE }}"
-            echo "Build env vars:"
-            echo "CARGO_INCREMENTAL=${{ env.CARGO_INCREMENTAL }}"
-            echo "CARGO_TARGET_DIR=${{ env.CARGO_TARGET_DIR }}"
-
-      - name: Run clippy
-        uses: actions-rs/clippy-check@v1.0.7
-        with:
-          # GitHub displays the clippy job and its results as separate entries
-          name: Clippy (stable) Results
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-features --all-targets -- -D warnings
-
-      # This check makes sure the crate dependency check is accurate
-      - name: Check Cargo.lock is up to date
-        uses: actions-rs/cargo@v1.0.3
-        with:
-          command: check
-          args: --locked --all-features --all-targets
-
-      # Edit zebra/deny.toml to allow duplicate dependencies
-      - name: Check for dependent crates with different versions
-        uses: EmbarkStudios/cargo-deny-action@v1.2.9
-        with:
-          command: check bans
-          args: --all-features --workspace
-
-      - name: Check crate sources
-        uses: EmbarkStudios/cargo-deny-action@v1.2.9
-        with:
-          command: check sources
-          args: --all-features --workspace
+    strategy:
+      matrix:
+        checks:
+          - bans 
+          - sources
 
-  fmt:
-    name: Rustfmt
-    timeout-minutes: 30
-    runs-on: ubuntu-latest
-    env:
-      CARGO_INCREMENTAL: 0
-      RUST_BACKTRACE: full
+    # Prevent sudden announcement of a new advisory from failing ci:
+    continue-on-error: ${{ matrix.checks == 'advisories' }}
```

Comment:
How will we notice advisories if they don't fail CI?

Reply:
The job will fail and it will show on the CI details. In the meanwhile, we might don't want to fix all the advisories until the team has the time to do it.

So it's like a soft fail. We will see the red signal on the specific job, but it won't stop the PR from getting merged.

Reply:
I'm not quite sure what I think about soft failures. But I don't want to block the rest of the PR on this.

In the past, soft failures have just piled up, and no-one actually deals with them. Soft failures can also be a bad experience for reviewers and external contributors, because it seems like the changes in the PR caused the failure.

Can we split the advisories into a separate PR, until we've worked out who will actually open these tickets?
(We also need to work out how to silence advisories we already have tickets for.)